libsolidity/analysis/TypeChecker.cpp

```diff
@@ -2533,8 +2533,15 @@ bool TypeChecker::visit(MemberAccess const& _memberAccess)
 	// TODO some members might be pure, but for example `address(0x123).balance` is not pure
 	// although every subexpression is, so leaving this limited for now.
 	if (auto tt = dynamic_cast<TypeType const*>(exprType))
+	{
 		if (tt->actualType()->category() == Type::Category::Enum)
 			annotation.isPure = true;
+		if (auto const* functionType = dynamic_cast<FunctionType const*>(annotation.type))
```

Comment:
Just looking at the diff here, I cannot easily see the relation between `exprType` and `anntation.type` - can you please clarify that?

Reply:
Actually, is `annotation.type` the resulting type? If possible, please avoid making any conclusions about the behaviour of an expression from the type it will result in.